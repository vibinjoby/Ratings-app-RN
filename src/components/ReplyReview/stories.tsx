import React from 'react'
import { storiesOf } from '@storybook/react-native'
import { View, StyleSheet } from 'react-native'

import ReplyReview from './component'

storiesOf('Components', module).add('ReplyReview', () => (
  <View style={styles.container}>
    <ReplyReview _id="123" onSend={() => console.log('send')} />
  </View>
))

const styles = StyleSheet.create({
  container: {
    padding: 20,
  },
})
