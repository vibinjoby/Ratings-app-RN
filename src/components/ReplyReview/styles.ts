import { StyleSheet } from 'react-native'
import Colors from '../../utilities/colors'
import Typography from '../../utilities/typography'

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  textInp: {
    ...Typography.RegularFont,
  },
  replyWrapper: {
    flex: 1,
    borderWidth: 1.5,
    borderRadius: 30,
    borderColor: Colors.borderColor,
    padding: 12,
    marginRight: 26,
  },
})
