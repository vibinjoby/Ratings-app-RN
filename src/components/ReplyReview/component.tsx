import React, { useState } from 'react'
import {
  Image,
  KeyboardAvoidingView,
  Platform,
  TextInput,
  TouchableOpacity,
  View,
} from 'react-native'

import styles from './styles'

export interface ReplyReviewProps {
  _id: string
  onSend: (arg0: string, arg1: string) => void
}

const ReplyReview: React.FC<ReplyReviewProps> = ({ _id, onSend }: ReplyReviewProps) => {
  const [inputVal, setInputVal] = useState('')
  return (
    <View style={styles.container}>
      <View style={styles.replyWrapper}>
        <TextInput
          value={inputVal}
          placeholder="Type Reply..."
          style={styles.textInp}
          onChangeText={(text) => setInputVal(text)}
        />
      </View>
      <TouchableOpacity activeOpacity={0.7} onPress={() => onSend(inputVal, _id)}>
        <Image source={require('../../assets/send/send.png')} />
      </TouchableOpacity>
    </View>
  )
}

export default ReplyReview
