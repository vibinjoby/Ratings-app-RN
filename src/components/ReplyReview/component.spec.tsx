import 'react-native'
import React from 'react'
import { render } from '@testing-library/react-native'

import ReplyReview from './component'

describe('ReplyReview', () => {
  it('should render', () => {
    const { toJSON } = render(<ReplyReview _id="123" onSend={() => console.log('send')} />)
    expect(toJSON()).toMatchSnapshot()
  })
})
