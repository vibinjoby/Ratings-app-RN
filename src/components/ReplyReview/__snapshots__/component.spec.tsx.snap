// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReplyReview should render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "borderColor": "#E2E2E2",
        "borderRadius": 30,
        "borderWidth": 1.5,
        "flex": 1,
        "marginRight": 26,
        "padding": 12,
      }
    }
  >
    <TextInput
      autoFocus={true}
      onChangeText={[Function]}
      placeholder="Type Reply..."
      style={
        Object {
          "fontFamily": "Poppins-Regular",
        }
      }
      testID="replyInput"
      value=""
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="sendBtn"
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/send/send.png",
        }
      }
    />
  </View>
</View>
`;
